### ISSUE TITLE
Resource Location Union Types Merging

### ISSUE THREAD

**Issue by Offroaders123** [Offroaders123/Region-Types#2]
I want to provide a way for users of the types to be able to extend the type definitions to include additional values they might be trying to access, which aren't part of the base game, nor it's types.

By that, an example I mean is that you should be able to define the types for what a given mod might add in terms of blocks. If the mod adds 10 new blocks to the game, that would mean that their Resource Location identifier types would be accessible through these APIs. If you don't represent those values in the type system, TS will get mad at you for trying to access them.

I'm not describing this very well, here's a little demo:

```ts
export type BlockResource =
  | "minecraft:acacia_wood"
  | "minecraft:activator_rail"
  | "minecraft:air"
  | "minecraft:allium"
  | "minecraft:amethyst_block"
  | "minecraft:amethyst_cluster"
  | "minecraft:ancient_debris"
  // ...and a lot more
;

export interface Chunk extends CompoundTag {
  sections: Section[];
  // ... other keys are also present
}

export interface Section extends CompoundTag {
  block_states: {
    palette: BlockState[];
  };
  // ...
}

export interface BlockState extends CompoundTag {
  Name: BlockResource;
  // ...
}
```

So say if the API is editing a modded world with custom blocks in it, you'd want to have the types reflect those values and include them in the union type.

If you had a custom ore for example, you might come across a block state with the `Name` attribute being `moreores:aluminium`. If you have a script set up to account for all block IDs or anything like that, then your checks wouldn't account for the possibility of encountering an Aluminium block. Similar to how you can install the type definitions for Node, Vite, Jest, things like that (In Node, the `Buffer` object exists on the global scope; TS doesn't know about that as part of the standard library, so Node provides typings to match the functionality it provides differently to that of the browser.), you could install the type definitions for mods, so you can get type checking on modded values as well.

In summary, I want to be able to extend the `BlockResource` type so users can provide their own values for it.

```ts
// Say if this were to merge with the vanilla definition above
export type BlockResource =
  | "moreores:aluminium"
  // ...and the Resource Location for other custom blocks that are added by the 'MoreOres' mod.
;
```
